### DISCUSSION TITLE
Multiple variables from IN filter

### DISCUSSION THREAD

**Discussion by RuBAN-GT** [prisma/prisma#14879] [Q&A]
Hi everyone,

I have a simple scenario: 
`findMany` request (to PG) with IN statement: 

```ts
where: {
  user_attribute: { some: { name: { equals: 'userID' }, value: { in: ['1', '2', '3'] } } },
}
```

This construction will be converted to SQL with three parameters-variables: 1, 2, 3 like `IN ($1,$2,$3)`. If I want to perform a filter with more values it will reserve separated variable per each IN element and generate huge SQL with expensive vars putting.

Is it ok from performance point of view? How I can combine these IN value to the one variable? Due injection protection I can't put into IN statement sth like `'1','2','3'`. 

Thank you!

**Comment by nurul3101** [comment]
HI @RuBAN-GT 👋 
What's the maximum number of values that you are going to pass in the `IN` clause? And also which database are you using?
There's a related issue in MariaDB where passing more than 999 elements in the `IN` statement causes an issue, see this related [comment](https://github.com/prisma/prisma/issues/12338#issuecomment-1195954336)

**Reply by RuBAN-GT** [reply]
Hi @nurul3101,
Thank you for your answer.
It will be less than 999 entries from PostgreSQL, but the question was is it ok, that each IN statement will passed as separated variable, is it time consuming operation to pass variables?
Thanks!

**Reply by nurul3101** [reply]
It shouldn't cause any issue, in case you feel the performance is not as expected you can enable [tracing](https://www.prisma.io/docs/concepts/components/prisma-client/opentelemetry-tracing) and debug the time consuming spans for a query.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
